src/ksm/client/discovery.go

```diff
@@ -34,11 +35,42 @@ const (
 	headlessServiceClusterIP = "None"
 )
 
-type lookupSRVFunc func(service, proto, name string) (cname string, addrs []*net.SRV, err error)
+type LookupSRVFunc func(service, proto, name string) (cname string, addrs []*net.SRV, err error)
+
+// DiscovererConfig holds parameters for creating discoverer.
+type DiscovererConfig struct {
+	LookupSRV         LookupSRVFunc
+	APIClient         client.Kubernetes
```

Comment:
We call this `K8sClient` in other places, is there a reason for not unifying those?

Reply:
Well, `discoverer` struct also has `apiClient`, but let's fix all that then. Pushed a fix :)